server/src/main/java/org/apache/cassandra/sidecar/cluster/CassandraAdapterDelegate.java

```diff
@@ -380,6 +411,23 @@ public NodeSettings nodeSettings() throws CassandraUnavailableException
         return nodeSettingsFromJmx;
     }
 
+    /**
+     * Returns the cached node settings read from the system_views.settings table. These are retrieved after a successful
+     * health check for native transport.
+     * @return Cached node settings from system_view.settings.
+     * @throws CassandraUnavailableException Thrown when native transport is not available. 
+     */
+    @Override
+    @NotNull
+    public Map<String, String> v2NodeSettings() throws CassandraUnavailableException
+    {
+        if (nodeSettingsFromCql == null)
+        {
+            throw new CassandraUnavailableException(CQL, "CQL NodeSettings unavailable");
+        }
+        return nodeSettingsFromCql;
```

Comment:
There's a chance for a race here, where one thread is retrieving the settings and another is updating settings to null in line 322 for example. I would recommend changing this code to avoid any potential race issues
```suggestion
        Map<String, String> settings = nodeSettingsFromCql;
        if (settings == null)
        {
            throw new CassandraUnavailableException(CQL, "CQL NodeSettings unavailable");
        }
        return settings;
```